projects/packages/connection-ui/src/class-admin.php

```diff
@@ -60,14 +62,16 @@ public function register_submenu_page() {
 	 */
 	public function enqueue_scripts( $hook ) {
 		if ( strpos( $hook, 'tools_page_wpcom-connection-manager' ) === 0 ) {
-			$build_assets = require_once __DIR__ . '/../build/index.asset.php';
-			wp_enqueue_script( 'jetpack_connection_ui_script', plugin_dir_url( __DIR__ ) . 'build/index.js', $build_assets['dependencies'], $build_assets['version'], true );
-
-			wp_set_script_translations( 'react-jetpack_connection_ui_script', 'jetpack' );
-			wp_add_inline_script( 'jetpack_connection_ui_script', $this->get_initial_state(), 'before' );
-
-			wp_enqueue_style( 'jetpack_connection_ui_style', plugin_dir_url( __DIR__ ) . 'build/index.css', array( 'wp-components' ), $build_assets['version'] );
-			wp_style_add_data( 'jetpack_connection_ui_style', 'rtl', plugin_dir_url( __DIR__ ) . 'build/index.rtl.css' );
+			Assets::register_script(
+				'jetpack_connection_ui',
+				'../build/index.js',
+				__FILE__,
+				array(
+					'in_footer' => true,
+				)
+			);
+			Assets::enqueue_script( 'jetpack_connection_ui' );
+			wp_add_inline_script( 'jetpack_connection_ui', $this->get_initial_state(), 'before' );
```

Comment:
Should we keep the translation handling in here?

```suggestion
			Assets::register_script(
				'jetpack_connection_ui',
				'../build/index.js',
				__FILE__,
				array(
					'in_footer' => true,
				)
			);
			Assets::enqueue_script( 'jetpack_connection_ui' );
			wp_add_inline_script( 'jetpack_connection_ui', $this->get_initial_state(), 'before' );
			wp_set_script_translations( 'jetpack_connection_ui', 'jetpack' );
```

Reply:
Should we fix the broken translation handling instead of deleting it? Sure, we can do that.

If my idea at https://github.com/Automattic/jetpack/issues/21690#issuecomment-966621370 pans out, I may wind up making "textdomain" an option to `Assets::register_script()` and have it complain if the script depends on wp-i18n but doesn't specify a domain.